src/pypilecore/common/piles/geometry/components/round.py

```diff
@@ -0,0 +1,435 @@
+        if show:
+            plt.show()
+        return axes
+
+    def plot_side_view(
+        self,
+        bottom_boundary_nap: float | Literal["pile_tip"] = "pile_tip",
+        top_boundary_nap: float | Literal["pile_head"] = "pile_head",
+        pile_tip_level_nap: float | int = -10,
+        pile_head_level_nap: float | int = 0,
+        figsize: Tuple[float, float] = (6.0, 6.0),
+        facecolor: Tuple[float, float, float] | str | None = None,
+        axes: Axes | None = None,
+        axis_arg: bool | str | Tuple[float, float, float, float] | None = "scaled",
+        show: bool = True,
+        **kwargs: Any,
+    ) -> Axes:
+        """
+        Plot the side view of the component at a specified depth.
+
+        Parameters
+        ----------
+        bottom_boundary_nap : float or str, optional
+            The bottom boundary level of the plot, in m w.r.t. NAP. Default = "pile_tip".
+        top_boundary_nap : float or str, optional
+            The top boundary level of the plot, in m w.r.t. NAP. Default = "pile_head".
+        pile_tip_level_nap : float, optional
+            The pile tip level in m w.r.t. NAP. Default = -10.
+        pile_head_level_nap : float, optional
+            The pile head level in m w.r.t. NAP. Default = 0.
+        figsize : tuple, optional
+            The figure size (width, height) in inches, by default (6.0, 6.0).
+        facecolor : tuple or str, optional
+            The face color of the pile cross-section, by default None.
+        axes : Axes
+            The axes object to plot the cross-section on.
+        axis_arg : bool or str or tuple, optional
+            The axis argument to pass to the `axes.axis()` function, by default "auto".
+        show : bool, optional
+            Whether to display the plot, by default True.
+
+        Returns
+        -------
+        Axes
+            The axes object to plot the cross-section on.
+        """
+        axes = instantiate_axes(
+            figsize=figsize,
+            axes=axes,
+            **kwargs,
+        )
+
+        if top_boundary_nap == "pile_head":
+            top_boundary_nap = pile_head_level_nap
+
+        if bottom_boundary_nap == "pile_tip":
+            bottom_boundary_nap = pile_tip_level_nap
+
+        (
+            component_head_level_nap,
+            component_tip_level_nap,
+        ) = self.get_component_bounds_nap(pile_tip_level_nap, pile_head_level_nap)
+
+        if (
+            top_boundary_nap > component_tip_level_nap
+            and bottom_boundary_nap < component_head_level_nap
+        ):
+            z_offset = component_head_level_nap
+            height = (
+                max(component_tip_level_nap, bottom_boundary_nap)
+                - component_head_level_nap
+            )
+
+            axes.add_patch(
+                patches.Rectangle(
+                    (self.cross_section_bounds[0], z_offset),
+                    self.diameter,
+                    height,
```

Comment:
if a `'primary_dimension': {'length': float}` is given can we add that to the plot as well.

For example when `height_base=1`. 
![image](https://github.com/user-attachments/assets/22e75579-f52e-4d13-b877-b11c9b546820)

Reply:
fixed